platform/src/default.rs

```diff
@@ -163,9 +77,16 @@ impl Platform for DefaultPlatform {
         if offset >= len {
             return Err(PlatformError::CertificateChainError);
         }
+
         let cert_chunk_range_end = min(offset + size, len);
-        out.copy_from_slice(&TEST_CERT_CHAIN[offset as usize..cert_chunk_range_end as usize]);
-        Ok(cert_chunk_range_end - offset)
+        let bytes_written = cert_chunk_range_end - offset;
+        if bytes_written as usize >= MAX_CHUNK_SIZE {
```

Comment:
If `bytes_written == MAX_CHUNK_SIZE` that should be ok. Just can't be larger.

Reply:
Done